arctic/store/version_store.py

```diff
@@ -517,6 +522,31 @@ def append(self, symbol, data, metadata=None, prune_previous_version=True, upser
         return VersionedItem(symbol=symbol, library=self._arctic_lib.get_name(), version=version['version'],
                              metadata=version.pop('metadata', None), data=None)
 
+    @mongo_retry
+    def update(self, symbol, data, **kwargs):
+        """
+        Find an existing chunk of data and update (overwrite) it. This WILL NOT
+        create a new version of the data! This is currently only supported by the
+        PandasDatetimeIndexedStore storage type.
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        data :
+            to be persisted
+        """
+        version = self._read_metadata(symbol)
+        old_ver = dict(version)
+        handler = self._update_handler(version, symbol, data, **kwargs)
+
+        if handler is None:
+            Exception("Update not implemented for handler %s" % handler)
+
+        mongo_retry(handler.update)(self._arctic_lib, version, symbol, data, **kwargs)
+        mongo_retry(self._versions.delete_one)(old_ver)
+        mongo_retry(self._versions.insert_one)(version)
```

Comment:
It does change. The version number doesnt, but the number of appended segments can change. If you update with a DF that requires new chunks to be added, it will update the existing ones and append the new ones.